mandatory dangerous extruder ?

Even if I don't use `PREVENT_DANGEROUS_EXTRUDE` it seams that `EXTRUDE_MINTEMP` is mandatory, otherwise : 

```
src/Marlin_main.cpp: In function ‘void gcode_M109()’:
src/Marlin_main.cpp:4135:43: error: ‘EXTRUDE_MINTEMP’ was not declared in this scope
.build/mega2560/Makefile:1100: recipe for target '.build/mega2560/src/Marlin_main.o' failed
```

The code causing the issue Marlin_main.cpp/4134: 

```
 // Prevents a wait-forever situation if R is misused i.e. M109 R0
  // Try to calculate a ballpark safe margin by halving EXTRUDE_MINTEMP
  if (degTargetHotend(target_extruder) < (EXTRUDE_MINTEMP/2)) return;
```

Having `EXTRUDE_MINTEMP` set is not a big problem, but maybe it should be made clear that it has to be set.

I havent tested, but I assume that since `PREVENT_DANGEROUS_EXTRUDE` is not set, I am still allowed to do cold extrude (basically a partial load filament - using bowden extruder)

--- a/Marlin/Conditionals.h
+++ b/Marlin/Conditionals.h
@@ -27,8 +27,19 @@
 
 #ifndef CONDITIONALS_H
 
+/**
+* Miscellaneous
+*/
 #ifndef M_PI
   #define M_PI 3.1415926536
+#endif
+
+/**
+ * This value is used by M109 when tying to calculate a ballpark safe margin
+ * to prevent wait-forever situation.
+ */
+#ifndef EXTRUDE_MINTEMP
+ #define EXTRUDE_MINTEMP 170
 #endif
 
 #ifndef CONFIGURATION_LCD // Get the LCD defines which are needed first

